cookbooks/cdo-nginx/templates/default/nginx.conf.erb

```diff
@@ -42,12 +42,21 @@ http {
     default "pegasus";
   }
 
+  map $http_x_cdo_backend $upstream_host {
+    # Override host using X-Cdo-Backend header.
+    dashboard "<%=CDO.dashboard_hostname%>";
+    pegasus "<%=CDO.pegasus_hostname%>";
```

Comment:
Note that using `CDO.` variables here implicitly depends on `cdo-secrets::app` recipe having been run before it. The latter also implicitly depends on `cdo-repository` having checked out the codebase which makes `deployment.rb` available to load. This works fine when run via `cdo-apps` (which includes both `cdo-repository` and `cdo-secrets`), but may break more isolated scenarios of running this cookbook/recipe directly.

Possible future cleanup would involve having `cdo-nginx` depend on `cdo-secrets`, and including `cdo-secrets::app` cookbook explicitly before using the CDO variable in this template.